src/controllers/movies.controller.ts

```diff
@@ -0,0 +1,203 @@
+import { type Request, type Response, type NextFunction } from 'express';
+
+import MovieModel, { type MovieBody } from '../models/movie.model';
+import {
+  type ReqParamsId,
+  type ReqParamsGenreName
+} from '../models/ReqParams.model';
+
+import initCustomError from '../utils/initCustomError';
+import { statusCodes, errorMessages } from '../utils/constants';
+
+class MoviesController {
+  static getAllMovies(req: Request, res: Response, next: NextFunction): void {
+    MovieModel.find()
+      .sort({ releaseDate: -1 })
+      .then((movies) => {
+        if (movies.length) {
+          res.status(statusCodes.OK).json(movies);
+        } else {
+          initCustomError(
+            next,
+            statusCodes.NOT_FOUND,
+            errorMessages.moviesListNotFound
+          );
+        }
+      })
+      .catch(() => {
+        initCustomError(
+          next,
+          statusCodes.NOT_FOUND,
+          errorMessages.moviesListNotFound
+        );
+      });
+  }
+
+  static createMovie(req: Request, res: Response, next: NextFunction): void {
+    const body = req.body as MovieBody;
```

Comment:
For improvements, the code will be looking better if you will use async/await syntax, without using callbacks:
`async createMovie(req: Request, res: Response, next: NextFunction): Promise<void> {
  try {
    const body = req.body as MovieBody;
    const title = body.title?.trim();
    const releaseDate = body.releaseDate;

    const duplicate = await MovieModel.findOne({
      title: new RegExp(`^${title}$`, 'i'),
      releaseDate
    });

    if (duplicate) {
      return initCustomError(next, statusCodes.CONFLICT, errorMessages.movieConflict);
    }

    const movie = await new MovieModel(body).save();
    res.status(statusCodes.CREATED).json(movie);
  } catch (err) {
    return initCustomError(next, statusCodes.BAD_REQUEST, err.message, err.name);
  }
}`

Reply:
Thank you! I will update my code accordingly.

Reply:
It is not in requirements, to use promises or async/await syntax, so, as you wish)

Reply:
Got it, thank you :)